src/Orleans.Core/Configuration/ServiceCollectionExtensions.cs

```diff
@@ -91,23 +92,20 @@ public static IServiceCollection AddGrainCallFilter(this IServiceCollection coll
         public static IServiceCollection UseStaticGatewayListProvider(this IServiceCollection collection,
             Action<StaticGatewayListProviderOptions> configureOptions)
         {
-            return collection
-                .Configure(configureOptions)
-                .AddSingleton<IGatewayListProvider, StaticGatewayListProvider>();
+            return collection.UseStaticGatewayListProvider(ob => ob.Configure(configureOptions));
         }
 
         /// <summary>
-        /// Add an <see cref="StaticGatewayListProvider"/> and configure it using <paramref name="configuration"/>
+        /// Add an <see cref="StaticGatewayListProvider"/> and configure it using <paramref name="configureOptions"/>
         /// </summary>
         /// <param name="collection"></param>
-        /// <param name="configuration"></param>
+        /// <param name="configureOptions"></param>
         /// <returns></returns>
         public static IServiceCollection UseStaticGatewayListProvider(this IServiceCollection collection,
-            IConfiguration configuration)
+            Action<OptionsBuilder<StaticGatewayListProviderOptions>> configureOptions)
         {
-            return collection
-                .Configure<StaticGatewayListProviderOptions>(configuration)
-                .AddSingleton<IGatewayListProvider, StaticGatewayListProvider>();
+            configureOptions?.Invoke(collection.AddOptions<StaticGatewayListProviderOptions>());
```

Comment:
given this can be null, should the param have a default value of `null`?

Reply:
Actually, scrap that - a default value would make overload resolution ugly.

Reply:
I'm not really making that call in this PR, just porting it. Although responding to this particular thing, it might be that we don't really want null, and potentially throw an argument exception... I don't really know at this point which behavior to choose, so I'll leave it undefined for now, unless you already thought through it and can recommend one